addons/uom/models/uom_uom.py

```diff
@@ -87,14 +88,24 @@ def _unprotected_uom_xml_ids(self):
     ]
 
     def _check_category_reference_uniqueness(self):
+        res_group = self.read_group(
+            [("category_id", "in", self.category_id.ids)],
+            ["category_id", "uom_type"],
+            ["category_id", "uom_type"],
+            lazy=False,
+        )
+        uom_by_category = defaultdict(int)
+        ref_by_category = {}
+        for res in res_group:
+            uom_by_category[res["category_id"][0]] += res["__count"]
+            if res["uom_type"] == "reference":
+                ref_by_category[res["category_id"][0]] = res["__count"]
```

Comment:
I wonder if something like
```suggestion
        categ_group = self.read_group(
            [("category_id", "in", self.category_id.ids), ('uom_type', '=', 'reference')],
            ["category_id"],
            ["category_id"],
            lazy=False,
        )
        categ_count = {categ['category_id'][0]: categ['__count'] for categ in categ_group}
```
Could work instead. :thinking: 
You don't really care about other `uom_type`, you only want to know if :
- You have at least one uom (given this check is triggered through the uom and not the category, I wonder if this auto-checks out? :thinking: )
- Your category *has* a reference (`category.id in categ_count)`
- Your category has only *one* reference (`categ_count[category.id] == 1`)

Reply:
That's what I wanted to do at first, but then I realized it wouldn't work when you don't have any **active** UoM in the category.
So it doesn't auto-checks out unfortunately :weary: 

Btw you described perfectly what the function does, and `categ_res` is definitely a better name than my generic `res_group` :laughing: